include/frg/cmdline.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef FRG_CMDLINE_HPP
+#define FRG_CMDLINE_HPP
+
+#include <frg/macros.hpp>
+#include <frg/string.hpp>
+#include <frg/tuple.hpp>
+
+#include <concepts>
+
+namespace frg FRG_VISIBILITY {
+
+struct option {
+	struct fn_type {
+		void (*ptr)(frg::string_view, void *);
+		void *ctx;
+		bool has_arg;
+	};
+
+	frg::string_view opt;
+	fn_type fn;
+
+	void apply(frg::string_view value) {
+		if (fn.ptr)
```

Comment:
in which valid usecase would `apply` be called while `fn.ptr` is null? i.e. are you meant to check the validity of an `option` object and only apply it while necessary, or should `apply` be called blindly without any checks?

Reply:
Good point. Seems like I had a bit of a brain fart there, you'd need to manually construct a `fn` that has a nulltptr in it, at which point you're already doing something wrong.

Reply:
I suppose we could add a `FRG_ASSERT` there though

Reply:
having some sort of check for this seems useful to catch bugs, but I'd say it should probably be an assert or something like that instead